doc/development/components.rst

```diff
@@ -0,0 +1,236 @@
+Main Components
+===============
+
+
+As introduced in the architecture overview, the FoamAdapter solver framework tries to make the solver more modular and easier to extend by breaking down the solver into smaller components.
+
+
+
+Main Concepts
+^^^^^^^^^^^^^
+
+The main concept of the FoamAdapter solver framework is to represent a CFD solver a series of operations that update fields based on governing equations.
+
+.. hint::
+
+   A CFD solver is a complex way to update fields with various operations based on governing equations.
+
+
+A context objects holds all relevant data for the solver, including fields and models and is passed to each operation that modifies the context.
+The context object is simply a simple data container that holds references to all relevant data structures:
+
+.. code-block:: python
+
+    class Context(BaseModel):
+        model_config = {"arbitrary_types_allowed": True}
+        fields: dict[str, Any]
+        models: dict[str, Any]
+        mesh: Any = None
+        runTime: Any = None
+
+
+Each operation is a class that implements a specific functionality, such as updating a field based on a governing equation or applying boundary conditions.
+It can be created by using the ``@Solver.step`` or ``@Model.step`` decorator.
```

Comment:
Is there a difference between `Solver.step` and `Model.step` if so what is the difference? 

Also, my golden rule about naming is, ideally stick with how you would describe it. You use the word `operation` and I think that is more intuitive then step.